urlchecker/core/urlproc.py

```diff
@@ -183,54 +273,24 @@ def check_urls(self, urls=None, retry_count=1, timeout=5):
         user_agent = get_user_agent()
         headers = {"User-Agent": user_agent}
 
-        # check links
-        for url in [url for url in urls if "http" in url]:
-
-            # init do retrails and retrails counts
-            do_retry = True
-            rcount = retry_count
-
-            # we will double the time for retry each time
-            retry_seconds = 2
-
-            # With retry, increase timeout by a second
-            pause = timeout
-
-            # No need to test the same URL twice
-            if url in seen:
-                continue
-
-            seen.add(url)
-            while rcount > 0 and do_retry:
-                response = None
-                try:
-                    response = requests.get(url, timeout=pause, headers=headers)
-
-                except requests.exceptions.Timeout as e:
-                    print(e)
-
-                except requests.exceptions.ConnectionError as e:
-                    print(e)
+        # check urls asyncronously
+        unique_urls = set([url for url in urls if "http" in url])
 
-                except Exception as e:
-                    print(e)
-
-                # decrement retrials count
-                rcount -= 1
-
-                # Break from the loop if we have success, update user
-                do_retry = check_response_status_code(url, response)
-
-                # If we try again, pause for retry seconds and update retry seconds
-                if rcount > 0 and do_retry:
-                    # keep this only for debugging
-                    # print("Retry n° %s for %s, with timeout of %s seconds." % (retry_count - rcount, url, pause))
-                    time.sleep(retry_seconds)
-                    retry_seconds = retry_seconds * 2
-                    pause += 1
-
-            # When we break from while, we record final response
-            self.record_response(url, response)
+        # handle different py versions support
+        if (3, 7) <= sys.version_info:
```

Comment:
What is the difference here?

Reply:
well I tested the code using python 3.6 and above. `asyncio.run()` was introduced in Python 3.7 and using the other alternative caused either warnings or errors so this was the best alternative imo to keep support for 3.6 and still provide a working code.

Reply:
Let's step back a second - what are the advantages of making this change? So far, I'm seeing that it introduces more potential for error because of the overlapping requests (leading to 429), adds an additional dependency (requests was just one), it hard to maintain between changing versions, and makes it impossible to use for Python < 3.5. Before you do a ton of work, we should seriously consider if this change is improving the library.

Reply:
Well there are clearly many drawbacks as you mention. The clear pro for this change is the huge speed difference. I cannot think of something else for now.